### DISCUSSION TITLE
[NodeJs] Prisma Connections Management for multiple Services

### DISCUSSION THREAD

**Discussion by dxgi** [prisma/prisma#14451] [Q&A] [answered]
Is it possible to reuse a single prisma connection for a multiple services NodeJs Rest API? I have about 5 services that will most likely use the database. All of them are completely separated so reusing a single Prisma/Client instance will not work.

+ How many connections am I gonna make if each services make their own connections?
+ Will the connections ever effect the performance of each connections?
+ Is there any way to reuse a prisma connection?

**Comment by pantharshit00** [comment, marked as answer]
Hello @dxgi, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks.


> How many connections am I gonna make if each services make their own connections?


By default, Prisma will spin up a connection pool with the limit of `num_physical_cpus` * 2 + 1. This has been taken from https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing. You can change the limit using the `?connection_limit` parameter in the connection string. 

> Will the connections ever effect the performance of each connections?



There should no impact if your database can handle that many connections. But it is always better to have as few connections as possible because acquiring a connection is expensive and can impact throughput. 

> Is there any way to reuse a prisma connection?



I am not sure what you mean here. Prisma uses a connection pool to manage connection so after the first acquisition, prisma reuses the connections from its connection pool.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
